aptos-move/framework/supra-framework/sources/pbo_delegation_pool.move

```diff
@@ -1601,10 +1586,237 @@ module supra_framework::pbo_delegation_pool {
+    /// will do so before the end of the transaction.
+    /// 
+    /// This function also assumes that `multisig_admin` is the admin of the `DelegationPool` at `pool_address`.
+    /// The caller must ensure that this is true.
+    fun lock_existing_delegators_stakes(
+        pool_address: address,
+        delegators: vector<address>,
+        stakes: vector<u64>
+    ) acquires DelegationPool {
+        let principle_stake_table =
+            &mut (borrow_global_mut<DelegationPool>(pool_address).principle_stake);
+
+        vector::zip_reverse(
+            delegators,
+            stakes,
+            |delegator, stake| {
+                if (stake > 0) {
+                    // Compute the stake to be added. `principle_stake` must be added to the table accordingly.
+                    if (table::contains(principle_stake_table, delegator)) {
+                        let stake_amount =
+                            table::borrow_mut(principle_stake_table, delegator);
+                        *stake_amount = *stake_amount + stake;
+                    } else {
+                        table::add(principle_stake_table, delegator, stake);
+                    };
+
+                    // Record the details of the lockup event. Note that only the newly locked
+                    // amount is reported and not the total locked amount.
+                    event::emit(
+                        UnlockScheduleApplied {
+                            pool_address,
+                            delegator,
+                            amount: stake
+                        }
+                    );
+                }
+                // else: The stake to be added is `0`. Do nothing.
+            }
+        );
+    }
+
+    /// Withdraws the specified `amount` from the `inactive` stake belonging to the given `delegator_address`
+    /// to the address of the `DelegationPool`'s `multisig_admin`, if available.
+    /// 
+    /// Note that this function is only temporarily intended to work as specified above and exists to enable The
+    /// Supra Foundation to ensure that the allocations of all investors are subject to the terms specified in the
+    /// corresponding legal contracts. It will be deactivated before the validator set it opened up to external 
+    /// validator-owners to prevent it from being abused.
+    fun admin_withdraw(
+        multisig_admin: &signer, pool_address: address, delegator_address: address, amount: u64
+    ) acquires DelegationPool, GovernanceRecords {
+        // Ensure that the caller is the admin of the delegation pool.
+        {
+            assert!(
+                is_admin(signer::address_of(multisig_admin), pool_address),
+                error::permission_denied(ENOT_AUTHORIZED)
+            );
+        };
+        assert!(amount > 0, error::invalid_argument(EWITHDRAW_ZERO_STAKE));
+        withdraw_internal(
```

Comment:
`withdraw_internal` can indeed fail since `redeem_shares` can fail therefore the precondition should be that whenever you call `admin_withdraw`, the amount does not exceed than what is available.

Reply:
Where is this precondition violated?